Source/JavaScriptCore/builtins/JSIteratorConstructor.js

```diff
@@ -71,3 +71,47 @@ function from(value)
     // 6. Return wrapper.
     return @wrapForValidIteratorCreate(iterator, iteratorNextMethod);
 }
+
+// https://tc39.es/proposal-iterator-sequencing/#sec-iterator.concat
+function concat(/* ...items */)
+{
+    "use strict";
+
+    var count = @argumentCount();
+
+    var openMethods = [];
+    var iterables = [];
+    for (var i = 0; i < count; ++i) {
+        var iterable = arguments[i];
+        if (!@isObject(iterable))
+            @throwTypeError("Iterator.concat expects all arguments to be objects");
+
+        var openMethod = iterable.@@iterator;
+        if (@isUndefinedOrNull(openMethod))
+            @throwTypeError("Iterator.concat expects all arguments to be iterable");
+
```

Comment:
Please add `@isCallable` check here: even tho `@call` with throw just the same TypeError in case of non-callable value, we should avoid performing `[[Get]]` on following arguments, as it is observable. It would be great to have a test for that.

Reply:
> It would be great to have a test for that.

in the test262 suite sometime later